File: test/src/main/java/com/shipping/domain/dto/LoginInfo.java
package com.shipping.domain.dto;


import com.shipping.domain.sys.User;
import org.spin.core.SpinContext;
import org.spin.core.auth.KeyInfo;
import org.spin.core.auth.TokenInfo;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

/**
 * token，密钥DTO
 * <p>Created by xuweinan on 2017/4/26.</p>
 *
 * @author xuweinan
 */
public class LoginInfo {
    private Long userId;
    private Map<String, Object> tokenInfo = new HashMap<>();
    private Map<String, Object> keyInfo = new HashMap<>();
    private UserDto userInfo;

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public Map<String, Object> getTokenInfo() {
        return tokenInfo;
    }

    public void setTokenInfo(TokenInfo tokenInfo, Long expiredIn) {
        this.tokenInfo.put("token", tokenInfo.getToken());
        this.tokenInfo.put("expiredSince", new Date(tokenInfo.getGenerateTime() + expiredIn));
    }

    public Map<String, Object> getKeyInfo() {
        return keyInfo;
    }

    public void setKeyInfo(KeyInfo keyInfo, Long expiredIn) {
        this.keyInfo.put("key", keyInfo.getKey());
        this.keyInfo.put("expiredSince", new Date(keyInfo.getGenerateTime() + expiredIn));
    }

    public UserDto getUserInfo() {
        return userInfo;
    }

    public void setUserInfo(User user) {
        this.userInfo = new UserDto(user);
    }
}

File: test/src/main/kotlin/com/shipping/domain/dto/LoginInfo.kt
package com.shipping.domain.dto


import com.shipping.domain.sys.User
import org.spin.core.auth.KeyInfo
import org.spin.core.auth.TokenInfo
import java.util.*

/**
 * token，密钥DTO
 *
 * Created by xuweinan on 2017/4/26.
 *
 * @author xuweinan
 */
class LoginInfo {
    var userId: Long? = null
    private val tokenInfo = HashMap<String, Any>()
    private val keyInfo = HashMap<String, Any>()
    var userInfo: UserDto? = null
        private set

    fun getTokenInfo(): Map<String, Any> {
        return tokenInfo
    }

    fun setTokenInfo(tokenInfo: TokenInfo, expiredIn: Long?) {
        this.tokenInfo["token"] = tokenInfo.token
        this.tokenInfo["expiredSince"] = Date(tokenInfo.generateTime!! + expiredIn!!)
    }

    fun getKeyInfo(): Map<String, Any> {
        return keyInfo
    }

    fun setKeyInfo(keyInfo: KeyInfo, expiredIn: Long?) {
        this.keyInfo["key"] = keyInfo.key
        this.keyInfo["expiredSince"] = Date(keyInfo.generateTime!! + expiredIn!!)
    }

    fun setUserInfo(user: User) {
        this.userInfo = UserDto(user)
    }
}
